napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -40,6 +39,18 @@
 
 # TODO: add error icon and handle pip install errors
 
+if TYPE_CHECKING:
+    import packaging.version
+
+
+def parse_version(v) -> 'packaging.version._BaseVersion':
+    import packaging.version
+
+    try:
+        return packaging.version.Version(v)
+    except packaging.version.InvalidVersion:
+        return packaging.version.LegacyVersion(v)
```

Comment:
Not sure if this is still a WIP, but should we define this in a util file somewhere and import it here. It seems fairly reusable?

Reply:
can if you want to... got an idea of where?

Reply:
i don't have strong feelings. `napari/utils/misc` works fine for me for now. can always then reorganize.